src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +144,1338 @@ static ColliderCylinderInit D_80B37A5C = {
+                EnWf_SetupSidestep(this, globalCtx);
+            } else if (shapeAngleDiff <= 10000) {
+                if (yawAngleDiff > 16000) {
+                    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+                    EnWf_SetupRunAroundPlayer(this);
+                } else {
+                    EnWf_ChangeAction(globalCtx, this, 1);
+                }
+            } else {
+                EnWf_SetupRunAroundPlayer(this);
+            }
+        }
+    }
+}
+
+void EnWf_SetupSecondSlash(EnWf* this) {
+    f32 endFrame = 1.0f;
+
+    if ((s32)this->skelAnime.curFrame >= 16) {
+        endFrame = 15.0f;
+    }
+
+    Animation_Change(&this->skelAnime, &gWolfosSlashingAnim, -0.5f, this->skelAnime.curFrame - 1.0f, endFrame,
+                     ANIMMODE_ONCE_INTERP, 0.0f);
+    this->action = WOLFOS_ACTION_SECOND_SLASH;
+    this->slashStatus = 0;
+    EnWf_SetupAction(this, EnWf_SecondSlash);
+}
+
+void EnWf_SecondSlash(EnWf* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s16 angle1 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+    s16 angle2 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    angle1 = ABS(angle1);
+    angle2 = ABS(angle2);
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (!Actor_IsFacingPlayer(&this->actor, 0x1554)) {
+            EnWf_SetupWait(this);
+            this->actionTimer = (Rand_ZeroOne() * 5.0f) + 5.0f;
+
+            if (angle2 > 0x32C8) {
+                this->headTilt = 30;
+            }
+        } else {
+            if ((Rand_ZeroOne() > 0.7f) || (this->actor.xzDistToPlayer >= 120.0f)) {
+                EnWf_SetupWait(this);
+                this->actionTimer = (Rand_ZeroOne() * 5.0f) + 5.0f;
+            } else {
+                this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+
+                if (Rand_ZeroOne() > 0.7f) {
+                    EnWf_SetupSidestep(this, globalCtx);
+                } else if (angle1 <= 0x2710) {
+                    if (angle2 > 0x3E80) {
+                        this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+                        EnWf_SetupRunAroundPlayer(this);
+                    } else {
+                        EnWf_ChangeAction(globalCtx, this, 1);
+                    }
+                } else {
+                    EnWf_SetupRunAroundPlayer(this);
+                }
+            }
+        }
+    }
+}
+
+void EnWf_SetupBackflip(EnWf* this) {
+    Animation_MorphToPlayOnce(&this->skelAnime, &gWolfosBackflippingAnim, -3.0f);
+    this->actor.speedXZ = -6.0f;
+    this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    this->actionTimer = 0;
+    this->isInvisible = true;
+    this->action = WOLFOS_ACTION_BACKFLIP;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_JUMP);
+    EnWf_SetupAction(this, EnWf_BackFlip);
+}
+
+void EnWf_BackFlip(EnWf* this, GlobalContext* globalCtx) {
```

Comment:
maybe, since this moves away from the player,

```suggestion
void EnWf_BackFlipAway(EnWf* this, GlobalContext* globalCtx) {
```

also to help distinguish it from `EnWf_TurnTowardsPlayer` / `EnWf_SomersaultAndAttack`